core/common/src/main/java/alluxio/cli/CommandUtils.java

```diff
@@ -120,22 +124,25 @@ public static void checkNumOfArgsNoMoreThan(Command cmd, CommandLine cl, int n)
    * @return list of the node names, null when file fails to read
    */
   @Nullable
-  public static List<String> readNodeList(String confDir, String fileName) {
+  public static Set<String> readNodeList(String confDir, String fileName) {
     List<String> lines;
+    String path = Paths.get(confDir, fileName).normalize().toString();
     try {
       lines = Files.readAllLines(Paths.get(confDir, fileName), StandardCharsets.UTF_8);
     } catch (IOException e) {
-      System.err.format("Failed to read file %s/%s. Ignored.", confDir, fileName);
-      return new ArrayList<>();
+      System.err.format("Failed to read file %s/%s. Ignored.%n", confDir, fileName);
+      return new HashSet<>();
     }
 
-    List<String> nodes = new ArrayList<>();
+    Set<String> nodes = new HashSet<>();
     for (String line : lines) {
       String node = line.trim();
       if (node.startsWith("#") || node.length() == 0) {
         continue;
       }
-      nodes.add(node);
+      if (nodes.add(node)) {
```

Comment:
Would `add()` return true when the set does not contain the element yet?

Reply:
My bad. I meant `if(!nodes.add(node))` where `nodes.add(node)` returns false if it's a duplicate